### PR TITLE
fix: de correction

### PR DESCRIPTION
**Pull request by cmeesters**
this PR fixes the issue of not correctly sorting and dropping log2FoldChange and padj columns for the downstream plots

It does NOT fix the statistical issues associated with insufficient data. Here, only optional imputation rules, will help.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new section for DESeq2 parameters in configuration files, enhancing organization.
	- Added support for dynamic output figure types based on configuration settings.

- **Improvements**
	- Increased CPU resource allocation for data analysis tasks to improve performance.
	- Enhanced parallel processing capabilities in workflow rules.
	- Streamlined configuration management in scripts for improved clarity and parameter usage.

- **Bug Fixes**
	- Updated output file paths to reflect new configuration settings, improving flexibility in file management.

- **Chores**
	- Refactored scripts to streamline configuration management and parameter usage.

### CODE REVIEW

**Comment by cmeesters** [comment]
forgot to add a revised CI version of the config file!

File: workflow/scripts/de_analysis.py [hunk 1]
```diff
@@ -36,29 +38,33 @@
...
 
 dds = DeseqDataSet(
     counts=counts_df,
     metadata=metadata,
-    design_factors=snakemake.config["design_factors"],
+    design_factors=snakemake.config["deseq2"]["design_factors"],
+    continuous_factors=snakemake.config["deseq2"].get("continuous_factors", None),
     refit_cooks=True,
     n_cpus=ncpus,
+    fit_type="mean",
```
[the remark this answers is not in the document]
Comment (cmeesters) [reply workflow/scripts/de_analysis.py:53]: right, will be fixed in the next push.

File: workflow/scripts/de_analysis.py [hunk 2]
```diff
@@ -117,33 +127,49 @@
 # normalized = normalized.join(np.array(pvalue[1]))
 normalized = normalized.join(padj)
 
-normalized.sort_values(by="log2FoldChange")
 # delete rows, which do not meet our p-value criterion
 # the comparison operator is >= because we drop all values >= our desired alpha
-normalized.drop(normalized[padj >= snakemake.config["alpha"]].index, inplace=True)
+# normalized.drop(normalized[padj >= snakemake.config["deseq2"]["alpha"]].index, inplace=True)
 normalized.to_csv(snakemake.output.normalized_counts)
+normalized.to_csv("normalized_my.csv")
```
[the remark this answers is not in the document]
Comment (cmeesters) [reply workflow/scripts/de_analysis.py]: done

File: workflow/scripts/de_analysis.py [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
 
 from snakemake.exceptions import WorkflowError
 
 from pydeseq2.dds import DeseqDataSet
 from pydeseq2.ds import DeseqStats
-from pydeseq2.utils import load_example_data
-
+from pydeseq2.default_inference import DefaultInference
 
 sys.stderr = sys.stdout = open(snakemake.log[0], "w")
```
[the remark this answers is not in the document]
Comment (cmeesters) [reply workflow/scripts/de_analysis.py:25]: no, that is the most practical solution

File: workflow/scripts/de_analysis.py [hunk 4]
```diff
@@ -155,39 +179,47 @@
...
         show=False,
         plotlegend=True,
         legendpos="upper right",
         legendanchor=(1.46, 1),
-        figtype="svg",
+        figtype=snakemake.config["deseq2"]["figtype"],
+    )
+    os.rename(
+        "volcano." + snakemake.config["deseq2"]["figtype"],
+        snakemake.output.volcano_plot,
```
[the remark this answers is not in the document]
Comment (cmeesters) [reply workflow/scripts/de_analysis.py]: - the significance level is only to allow fake ci runs.
- yes, tuples are expected for lfc_thr and pv_thr
- the rename is only for fake ci runs.

**Review by yeising — APPROVED** [review]

**Pull request merged**